### PR TITLE
refactor youtube API integration & add tests

### PR DESCRIPTION
**Pull request by dereckquock**
- refactor youtube integration
  - split out each API call into its own function
  - extract the sorting function
  - use `maxResults` instead of tail recursion
- add tests with mocks
- use a `<View>` instead of `<FlatList>` within a `<ScrollView>` for the `<Playlist>`
- remove showing the publish date for the playlist videos because they don't seem to be correct and don't correspond to what's shown in youtube
  > i'm not sure how valuable this is anyways 🤔

### CODE REVIEW

File: data/__tests__/youtube.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    { id: '3', publishDate: '2020-04-10T00:00:00Z' },
+    { id: '1', publishDate: '2020-04-01T00:00:00Z' },
+    { id: '4', publishDate: '2020-04-15T00:00:00Z' },
+  ];
+
+  expect(items.sort(sortByDate)).toMatchSnapshot();
+});
+
+test('should fetch channel sections', async () => {
+  axios.get.mockResolvedValue({
```
Comment (dereckquock) [inline data/__tests__/youtube.js:19]: we should probably put mocks in a separate file, but this is fine for now

File: data/youtube.js [hunk 2]
```diff
@@ -1,14 +1,11 @@
 import axios from 'axios';
 import Keys from '../constants/Keys';
 
-const TARGET_SECTION_TITLE = "Popular Series"
-
-const PLAYLIST_PARTS = ['id', 'contentDetails', 'snippet'];
```
Comment (dereckquock) [inline data/youtube.js:6]: each call doesn't require all of these parts, so we can just explicitly define them in the params

File: navigation/AppNavigator.js [hunk 3]
```diff
@@ -59,7 +59,14 @@ const defaultOptions = {
     />
   ),
   headerTitle: (props) => (
-    <Text {...props} L allowFontScaling={false}>
+    <Text
+      {...props}
+      L
+      adjustsFontSizeToFit
+      allowFontScaling={false}
+      numberOfLines={1}
```
Comment (dereckquock) [inline navigation/AppNavigator.js:67]: allow the header to change its font size to fit within the single line

File: screens/Playlist.js [hunk 4]
```diff
@@ -32,26 +30,17 @@ const storePlaylistData = async (playlistID, data) => {
...
-    playlistDescription,
-    playlistURI,
-  } = route.params;
-  console.log(route.params);
-  navigation.setOptions({ title: playlistTitle });
   const insets = useSafeArea();
-  const ref = React.useRef(null);
+  const { playlistID, playlistTitle, playlistURI } = route.params;
 
-  useScrollToTop(ref);
```
Comment (dereckquock) [inline screens/Playlist.js:54]: we don't need to handle this for this screen. this is only used for the main screens that aren't children of a stack

File: screens/Playlist.js [hunk 5]
```diff
@@ -166,18 +146,11 @@ const PlaylistScreen = ({ navigation, route }) => {
                       style={styles.thumbnailStyle}
                       resizeMode="cover"
                     />
-                    <View style={{ flex: 1, flexDirection: 'column' }}>
-                      <Heading style={styles.title}>{title}</Heading>
-                      <Text style={[styles.description, { paddingTop: 8 }]}>
-                        {new Date(publishDate).toLocaleDateString()}
-                      </Text>
```
Comment (dereckquock) [inline screens/Playlist.js:173]: we probably don't _need_ to show the date...it was kinda more confusing than useful cause it's not necessarily when the message was given

File: data/youtube.js [hunk 6]
```diff
@@ -24,159 +21,119 @@ axios.interceptors.response.use(response => {
...
+ */
+export async function fetchPlaylistItems(playlistId) {
   const { data = {} } =
     (await axios.get(`${url}/playlistItems`, {
       params: {
-        part: PLAYLIST_PARTS.join(),
-        playlistId,
         key: API_KEY,
-        pageToken: nextPage,
+        maxResults: 30, // ? 🤔 not sure the max # of videos in series playlists
```
Comment (Phillip3) [inline data/youtube.js:113]: @dereckquock it's like 4 videos :)

**Review by Phillip3 — APPROVED** [review]
@dereckquock you're amazing and I'm in awe.

File: data/youtube.js (hunk 6 above)
Comment (dereckquock) [reply data/youtube.js:113]: i think sometimes, there's like 7 videos 😅  i'm adding some buffer for times when maybe a series is a month long 😂

**Pull request merged**